### PR TITLE
Heavy rework on documentation

### PR DESCRIPTION
**Pull request by legacycode**
Hi,
i did a lot of restructuring the existing documentation. The documentation now automatically renders on readthedocks.io. I think this is a great new user experience :-)

I hope you like my suggestions. You can preview the new documentation at: https://btcd.readthedocs.io/

I like this kind of documentation. That was a lot of work. Hopefully this will be merged :-)
/legacycode

### CODE REVIEW

**Comment by bjornoj** [comment]
readthedocs is very nice, maybe add a link to it in the README?

I think it would be nice to  delete docs/table_of_contents.md and rename docs/index.md to docs/README.md because README.md is a natural starting point, index is not

**Comment by legacycode** [comment]
@jakesyl done

**Comment by jakesyl** [comment]
Thanks!

**Comment by jakesyl** [comment]
@jcvernaleo (as per #1530)

- low priority
- documentation

This probably does make sense to merge sooner rather than later since I've already had @legacycode rework this once and new PRs will likely cause conflicts that he'll have to deal with again

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+import recommonmark
+from recommonmark.transform import AutoStructify
+
+# -- Project information -----------------------------------------------------
+
+project = 'btcd'
+copyright = '2019, btcd'
+author = 'btcd'
```
Comment (onyb) [inline docs/conf.py:23]: Let's update the year.

Also, is it possible to set the author as "btcsuite developers"?

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+# The full version, including alpha/beta/rc tags
+release = 'beta'
+
+source_suffix = ['.rst', '.md']
```
Comment (onyb) [inline docs/conf.py:28]: If we're not using using `.rst` files, can we remove it?

File: docs/code_contribution_guidelines.md [hunk 3]
```diff
@@ -1,23 +1,4 @@
...
-5.2 [Rework Code (if needed)](#CodeRework)<br />
-5.3 [Acceptance](#CodeAcceptance)<br />
-6. [Contribution Standards](#Standards)<br />
-6.1. [Contribution Checklist](#Checklist)<br />
-6.2. [Licensing of Contributions](#Licensing)<br />
-
-<a name="Overview" />
-
-### 1. Overview
+# Code contribution guidelines
```
Comment (onyb) [inline docs/code_contribution_guidelines.md:1]: What do you think about using a shorter name like `contributing.md`?

File: docs/configuring_tor.md [hunk 4]
```diff
@@ -1,25 +1,4 @@
...
-4.3 [Config File Example](#BridgeConfigFileExample)<br />
-5. [Tor Stream Isolation](#TorStreamIsolation)<br />
-5.1 [Description](#TorStreamIsolationDescription)<br />
-5.2 [Command Line Example](#TorStreamIsolationCLIExample)<br />
-5.3 [Config File Example](#TorStreamIsolationFileExample)<br />
-
-<a name="Overview" />
-
-### 1. Overview
+# Configuring TOR
```
Comment (onyb) [inline docs/configuring_tor.md:1]: Shouldn't this content belong to  `configuration.md`?

File: docs/developer_resources.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Developer Resources
+
+* [Code Contribution Guidelines](https://github.com/btcsuite/btcd/tree/master/docs/code_contribution_guidelines.md)
```
Comment (onyb) [inline docs/developer_resources.md:3]: Make sure you update this if you change the filename.

File: docs/update.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Update
```
Comment (onyb) [inline docs/update.md:1]: Not sure if this deserves a separate markdown file. How about adding it to the installation docs?

File: docs/json_rpc_api.md [hunk 7]
```diff
@@ -1,4 +1,5 @@
-### Table of Contents
+# JSON RPC API
```
Comment (onyb) [inline docs/json_rpc_api.md:1]: This file could be renamed as `json-rpc.md` or `rpc.md`.

File: docs/configuring_tor.md (hunk 4 above)
Comment (legacycode) [reply docs/configuring_tor.md:1]: i think tor is a complicated thing and the normal configuration.md should focus on the default configuration. i would like to suggest keeping it in an own file.

**Comment by onyb** [comment]
@legacycode Any update on this? The CI needs to be fixed in order to merge this.

**Comment by jakesyl** [comment]
That and needs to be rebased

**Comment by legacycode** [comment]
@onyb @jakesyl Rebased and CI passed. Ready for merge :-)

**Comment by jakesyl** [comment]
LGTM!

**Comment by legacycode** [comment]
@jakesyl Great. I released the subdomain btc.readthedocs.org on https://readthedocs.org/dashboard/ Please make sure that somebody of your team create/administrate the project an readthedocs.io. It's a one time process.

**Pull request merged**